src/main/java/seedu/address/logic/commands/NegateCommand.java

```diff
@@ -0,0 +1,29 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.core.index.Index;
+
+/**
+ * Represents a command with hidden internal logic and the ability to reverse the execution of other commands.
+ */
+public abstract class NegateCommand extends Command {
```

Comment:
does it have to be abstract since this class doesn't contain any abstract methods.. I think a normal class would suffice